wire/msgtx.go

```diff
@@ -589,6 +596,12 @@ func (msg *MsgTx) btcDecode(r io.Reader, pver uint32, enc MessageEncoding,
 				sbuf = sbuf[len(txin.Witness[j]):]
 			}
 		}
+
+		// Check that if the witness flag is set that we actually have
```

Comment:
Nice, this is something that _is_ actually spelled out in the old BIP (144): 
> If the witness is empty, the old serialization format must be used.

https://github.com/bitcoin/bips/blob/master/bip-0144.mediawiki